HashToSlashCommentFixer - Add

SpacePossum opened the pull request:
Just a little fixer from my collection I think others might want to use once in a while (so on contrib level).
It changes single line comments starting with `#` to starting with `\\`.
PHP handles those the same so changing those should not be a problem.
For example

```php
<h1>This is an <?php #echo 123;?> example</h1>
```
to
```php
<h1>This is an <?php //echo 123;?> example</h1>
```

```php
<?php
    # test
```
to
```php
<?php
    // test
```

gharlan commented:
> I think others might want to use

:+1: Thanks for this one.

gharlan commented on README.rst:
```diff
@@ -577,6 +577,11 @@ Choose from the list of available fixers:
 
+* **hash_to_back_slash_comment** [contrib]
+                        Single line comments should
+                        use double backslash (//) and
```
`//` are not backslashes ;)
So the fixer name is wrong, too.

SpacePossum replied:
oh man... it is Friday right?

SpacePossum replied:
maybe `hash_to_slashes_comment` ?

SpacePossum replied:
renamed ;)

gharlan commented:
symfony level?

keradus commented:
multiline test should be added:
```php
            array(
                '<?php
                    // foo
                    // bar
                ',
                '<?php
                    # foo
                    # bar
                ',
            ),
```

keradus commented:
Also, I think that this should also be changed by fixer:
```php
            array(
                '<?php
                    // foo
                ',
                '<?php
                    ## foo
                ',
            ),
```

------------------

Perhaps even this:
```php
            array(
                '<?php
                    ////////// foo
                ',
                '<?php
                    ########## foo
                ',
            ),
```

keradus commented:
> symfony level?

yes

SpacePossum commented:
Test added, rebased, moved to Symfony, squashed

I disagree with https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1610#issuecomment-167425472 so I won't change it, I thing we should not touch the content of comments (and even if this should be done, than it should not be done with this fixer)

keradus commented:
:+1:

keradus commented:
Thank you @SpacePossum.

SpacePossum commented:
thank you @keradus :+1:
